### PR TITLE
Added support for ignoring certain cookies from executing a preflight…

### PR DESCRIPTION
**Pull request by Toflar**
… request

### CODE REVIEW

**Review by ausi — APPROVED** [review]
LGTM

**Review by aschempp — COMMENTED** [review]
Why did not not just add a second constructor argument? It would make it way more obvious what "options" are available?

Also, because options cannot be extended at runtime, it's actually not necessary to have an array of regular expressions. One could simply write one expression to match all cookies (like so: `/^Foo$|^Bar$/`), right @ausi ?

File: src/SymfonyCache/HeaderReplaySubscriber.php [hunk 1]
```diff
@@ -147,4 +162,31 @@ private function checkRequest(Request $request)
...
+        $count = $cookies->count();
+
+        if (0 === count($this->options['ignore_cookies'])) {
+            return 0 !== $count;
+        }
+
+        foreach ($cookies as $name => $value) {
+            foreach ($this->options['ignore_cookies'] as $rgxp) {
+                if (!preg_match($rgxp, $name)) {
+                    return true;
```
Comment (ausi) [inline src/SymfonyCache/HeaderReplaySubscriber.php]: Does this actually work? If I have two `ignore_cookies`, both of them would have to match every cookie with this code I think.

**Review by ausi — CHANGES_REQUESTED** [review]

File: src/SymfonyCache/HeaderReplaySubscriber.php (hunk 1 above)
Comment (Toflar) [reply src/SymfonyCache/HeaderReplaySubscriber.php]: True, fixed and added quite some more unit test cases in https://github.com/terminal42/header-replay-bundle/pull/5/commits/b25135ea12e816a3fb53e4dd28e77c07eb51b769.

**Pull request merged**